Client/Frontend/Reader/ReadabilityService.swift

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+
+private let ReadabilityServiceSharedInstance = ReadabilityService()
+
+private let ReadabilityTaskDefaultTimeout = 15
+private let ReadabilityServiceDefaultConcurrency = 1
+
+enum ReadabilityOperationResult {
+    case Success(ReadabilityResult)
+    case Error(NSError)
+    case Timeout
+}
+
+class ReadabilityOperation: NSOperation, WKNavigationDelegate, ReadabilityBrowserHelperDelegate {
+    var url: NSURL
+    var semaphore: dispatch_semaphore_t
+    var result: ReadabilityOperationResult?
+
+    init(url: NSURL) {
+        self.url = url
+        self.semaphore = dispatch_semaphore_create(0)
+    }
+
+    override func main() {
+        if self.cancelled {
+            return
+        }
+
+        // Setup a browser, attach a Readability helper.
+
+        let configuration = WKWebViewConfiguration()
+        let browser = Browser(configuration: configuration)
```

Comment:
Also, I'm not sure this is safe to do. According to the [Cocoa thread safety summary](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/Multithreading/ThreadSafetySummary/ThreadSafetySummary.html): 

> The `NSView` class is generally not thread-safe. You should create, destroy, resize, move, and perform other operations on `NSView` objects only from the main thread of an application.